C#/Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -207,38 +205,56 @@ public object ActivityToSlack(Activity activity)
         /// <param name="activities">An array of outgoing activities to be sent back to the messaging API.</param>
         public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
         {
-            ResourceResponse[] responses = { };
+            List<ResourceResponse> responses = new List<ResourceResponse>();
             for (var i = 0; i < activities.Length; i++)
             {
                 Activity activity = activities[i];
                 if (activity.Type == ActivityTypes.Message)
                 {
-                    dynamic message = ActivityToSlack(activity as Activity);
+                    dynamic message = ActivityToSlack(activity);
 
                     try
                     {
-                        SlackAPI slack = await this.GetAPIAsync(turnContext.Activity);
-                        ChatPostMessageResult result = null;
+                        SlackTaskClient slack = await this.GetAPIAsync(turnContext.Activity);
+                        ChatPostMessageResult result = new ChatPostMessageResult();
 
                         if (message.ephemeral)
                         {
-                            // result = await slack.Chat // TODO
+                            Console.WriteLine("Post Ephemeral:", message);
```

Comment:
I am not sure if we should use `console`, as it depends on what kind of project the final user will use.

Reply:
good point